pkg/virt-controller/watch/migration/migration.go

```diff
@@ -244,6 +244,9 @@ func (c *Controller) runWorker() {
 }
 
 func (c *Controller) Execute() bool {
+	if c.Queue.Len() == 0 {
```

Comment:
So this would prioritize the work we need but once we flush all pending migrations we are at square 0 as it will take some time to re-evaluate all pending migrations. 
I believe this would not be enough to fix the issue described.

Reply:
Yes, we don't re-evalutate all pending migrations in a hot loop anymore, but we still do every time a migration finishes (or when the main queue is empty, which is an edge-case).
I believe it greatly improves the situation, but it could indeed not be enough... Maybe then we can increase the number of threads?
If you have any other idea, please chime in! I agree my solution is not ideal, I just couldn't come up with a better one...

Reply:
1. priority queue, where objects that are stuck on pending will be enqueued with lower priority, `Get` will prefer keys with higher priority so you always process active migration first.
2. Split the controller into two subroutines/controllers, where one is responsible for admission and the second would do everything else.  The admission routine can run with one thread/goroutine as it would require lock anyway.